### ISSUE TITLE
Question about training custom data

### ISSUE THREAD

**Issue by fxcdl** [eladrich/pixel2style2pixel#280]
Hi, thanks for your code
I want to train PSP on my own paired dataset, and the dataset is independent of the face.
After  I read README and some related questions,  I made the following changes:
![image](https://user-images.githubusercontent.com/50817018/176109688-47fbff0e-b180-4e3b-aac9-e8c56c6fafde.png)
![image](https://user-images.githubusercontent.com/50817018/176109754-badd1587-5d73-45f9-80f1-a8b8be5141f2.png)
then, I successfully run the training code:
![image](https://user-images.githubusercontent.com/50817018/176110303-05a65da1-0def-4fc3-b6c0-b84941fcd412.png)

My model is still in training. I have some questions:
1. Although the code is running, I'm not sure if my settings are correct.
2. I don't understand  the difference between "training the PSP encoder" and other orders,  do I only need to use the "training the PSP encoder"  order on my dataset.

**Comment by yuval-alaluf** [comment]
The settings look fine. I would recommend running with more than 0 workers in order to speed up training time considerably. 
If you wish to train an encoder (i.e, perform inversion), then you are correct that you want to follow the "training the pSp encoder" portion of the README.

**Comment by fxcdl** [comment]
OK，thanks for your reply!

**Comment by fxcdl** [comment]
Hi, i trained an encoder on my datasets, and i think the result is abnormal.
![499700](https://user-images.githubusercontent.com/50817018/179454826-d34b9bcf-9cec-42ef-b797-1cb131c74acc.jpg)
this is the result on training datasets which has 556 pair of images.
Is my training set too small?

**Comment by yuval-alaluf** [comment]
Hi @fxcdl ,
It seems like you tried uploading some images but the images were not uploaded.
In general, 500 images does seem a bit low for training these types of models.

**Comment by fxcdl** [comment]
> Hi @fxcdl , It seems like you tried uploading some images but the images were not uploaded. In general, 500 images does seem a bit low for training these types of models.
![1](https://user-images.githubusercontent.com/50817018/180106227-26c85e90-62f8-40ce-b36a-4b6668e90596.png)
aha, is it uploading now?

**Comment by yuval-alaluf** [comment]
One option that can help here is to add the moco-based loss to training (using `moco_lambda`) and if that doesn't help, you can try strong inversion techniques such as restyle, hyperstyle, or PTI, all of which are available and compatible with SG2.
